test/regressions/index.test.js

```diff
@@ -101,29 +101,11 @@ async function main() {
         // Move cursor offscreen to not trigger unwanted hover effects.
         page.mouse.move(0, 0);
 
-        const pathsToNotWaitForFlagCDN = [
-          '/docs-data-grid-filtering/HeaderFilteringDataGridPro', // No flag column
-          '/docs-data-grid-filtering/CustomHeaderFilterDataGridPro', // No flag column
-          '/docs-data-grid-filtering/CustomHeaderFilterSingleDataGridPro', // No flag column
-          '/docs-data-grid-filtering/SimpleHeaderFilteringDataGridPro', // No flag column
-          '/docs-data-grid-filtering/ServerFilterGrid', // No content rendered
-          '/docs-data-grid-filtering/CustomMultiValueOperator', // No content rendered
-          '/docs-data-grid-filtering/QuickFilteringInitialize', // No content rendered
-          '/docs-data-grid-sorting/FullyCustomSortComparator', // No flag column
-          '/docs-data-grid-sorting/ServerSortingGrid', // No flag column
-          '/docs-data-grid-filtering/QuickFilteringExcludeHiddenColumns', // No flag column
-          '/docs-data-grid-filtering/QuickFilteringDiacritics', // No flag column
-        ];
-
-        if (
-          /^\/docs-data-grid-(filtering|sorting)/.test(pathURL) &&
-          !pathsToNotWaitForFlagCDN.includes(pathURL)
-        ) {
-          // Wait for the flags to load
-          await page.waitForResponse((response) =>
-            response.url().startsWith('https://flagcdn.com'),
-          );
-        }
+        // Wait for the flags to load
+        await page.waitForFunction(() => {
+          const images = Array.from(document.querySelectorAll('img'));
+          return images.every((img) => img.complete);
+        });
```

Comment:
This should work better now

Reply:
Nice, even simpler than https://github.com/mui/mui-x/pull/5473#issuecomment-1339924982, relying on pulling.